iheart-songs/src/songs-table.tsx

```diff
@@ -1,11 +1,40 @@
-import React, { useMemo } from "react";
+import React, { useEffect, useMemo, useState } from "react";
 import "./songs-table.css";
 import useSongs from "./hooks/use-songs";
 import { useHistory } from "react-router";
+import { Song } from "./API";
+import loadingGif from "./assets/images/loading.gif";
 
 const SongsTable: React.FC = () => {
   const [loading, _, songs] = useSongs();
-  
+  const [songsData, setSongsData] = useState<Song[]>(songs || []);
+  const [isAscending, setIsAscending] = useState<boolean>(true);
+  const [sortedColumnHeader, setSortedColumnHeader] = useState("");
+  const UP_ARROW = "🔼";
```

Comment:
not sure whether using emoji like this is right approach. for now, let's proceed